common/test/acceptance/tests/lms/test_learner_profile.py

```diff
@@ -159,10 +159,16 @@ def verify_profile_forced_private_message(self, username, birth_year, message=No
         """
         Verify age limit messages for a user.
         """
-        self.set_birth_year(birth_year=birth_year if birth_year is not None else "")
+        if birth_year:
```

Comment:
The problem with the `foo if test else bar` idiom in this case, is that it's trying to jam a LOT of logic into one line. It becomes hard to understand what is happening. Let me try something different. 

Also, your comment about 0 as falsy makes sense. I'll change that.